helper/identity/templating.go

```diff
@@ -102,16 +122,35 @@ func performTemplating(ns *namespace.Namespace, input string, entity *Entity, gr
 		case trimmed == "id":
 			return entity.ID, nil
 		case trimmed == "name":
-			if entity.Name == "" {
+			if entity.Name == "" && !allowMissingSelectors {
```

Comment:
I wonder if it's too broad to apply a single allow missing selectors bool across all cases. It's probably fine for now but we should keep our eye out in the future for whether we might need to change it.

Reply:
Switch to mode which should buy us some flexibility.